ambari-server/src/main/java/org/apache/ambari/server/controller/AddServiceRequest.java

```diff
@@ -172,20 +196,39 @@ public Configuration getConfiguration() {
     return ConfigurableHelper.convertConfigToMap(configuration);
   }
 
+  @JsonIgnore // TODO confirm: credentials shouldn't be serialized
+  @ApiIgnore
+  public Map<String, Credential> getCredentials() {
+    return credentials;
+  }
+
+  @JsonIgnore
+  @ApiIgnore
+  public Optional<SecurityConfiguration> getSecurity() {
+    return Optional.ofNullable(security);
+  }
+
+  @JsonProperty(SECURITY)
+  @ApiModelProperty(name = SECURITY)
+  public SecurityConfiguration _getSecurity() {
```

Comment:
Minor issue:  This method doesn't appear to be used in this current patchset.  Why are there two methods for returning the SecurityConfiguration object?  

Would it be better to just use the form that returns the Optional<SecurityConfiguration>?

Reply:
Yes, JSON export and Swagger.  It's similar to the existing `getConfigurationContents()` vs. `getConfiguration()` methods, where we have separate methods for JSON and internal usage.

Reply:
Great.  Thanks for clarifying this.